project/app/api/submission.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from fastapi import APIRouter, File, UploadFile
+from pydantic import BaseModel, Field, validator
+
+router = APIRouter()
+log = logging.getLogger(__name__)
+
+
+class Submission(BaseModel):
+    """Data Model to parse Request body JSON, using the Default values
+    will check if the s3 API server is currently accepting requests"""
+
+    story_id: str = Field(..., example="12345")
+    story_file: bytes = Field(..., example=b"10gh1r447/s4413.")
+
+    @validator("story_id")
+    def no_none_ids(cls, value):
+        """Ensure that the value is not passed as None"""
+        assert value is not None
+        return value
+
+    @validator("story_file")
+    def check_file(cls, value):
+        # design method to validate file contents
+        # XXX: (passing a hash then comparing that?)
+        # assert hashlib.sha256(data=file).hexdigest() == passed_hash
+        return value
+
+
+@router.post("/submission/text")
+async def submission_text(sub: Submission):
+    """Function that takes a file that is sent via FastAPI FileUpload interface
+    and uploads that file to an s3 bucket, then returns the link for the s3
+    bucket object and if the content has been flagged as inappropriate content.
```

Comment:
Has the file been transcribed to text at this point? If not, I think the text moderation step would need to come after it has been transcribed. If the transcription happens in this step can you mention it in the doc string? It was a little unclear to me

Reply:
ok I wasn't too sure how I was going to go about that, my plan was to have two async services, one that handles the s3 bucket upload, and one that handles the transcription and RDS entries. this method would call both services and wait for their responses then return that status to web with the link to the s3 bucket.

Reply:
Okay, then it is probably fine as-is for now. When you decide how you want to set it up and implement those pieces, just make sure to come back and clarify this.